src/payment/strategies/bluesnapv2/bluesnapv2-widget-loader.ts

```diff
@@ -0,0 +1,32 @@
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { PaymentMethodCancelledError } from '../../errors';
+
+import { BlueSnapV2PaymentInitializeOptions } from './bluesnapv2-payment-options';
+
+export default class BlueSnapV2WidgetLoader {
+    load(frameName: string, options?: BlueSnapV2PaymentInitializeOptions) {
+        return new Promise((_, reject) => {
+            if (!options) {
+                throw new NotInitializedError(
+                    NotInitializedErrorType.PaymentNotInitialized
+                );
+            }
+
+            const { onLoad } = options;
+            onLoad(undefined, this._createFrame(frameName), () => {
+                reject(new PaymentMethodCancelledError());
+            });
+        });
+    }
+
+    private _createFrame(name: string): HTMLIFrameElement {
+        const iframe = document.createElement('iframe');
+
+        iframe.name = name;
```

Comment:
how does this work? doesnt the iframe need a src value?

Reply:
No, it doesn't. I'm loading the content through the name of the iframe, by setting the form's target to the iframe's name.

https://github.com/bigcommerce/form-poster-js/blob/e0bc8d700d2db7ab02f2e9a0d7e36f34f3c578b7/src/form-builder.ts#L9